src/py/mat3ra/made/tools/analyze/fingerprint/layered_fingerprint_along_axis.py

```diff
@@ -0,0 +1,113 @@
+from itertools import cycle, islice
+from typing import List, Sequence
+
+from mat3ra.esse.models.core.reusable.axis_enum import AxisEnum
+from pydantic import BaseModel, Field
+
+from .layer_fingerprint import LayerFingerprint
+
+
+def _jaccard(a: Sequence[str], b: Sequence[str]) -> float:
```

Comment:
This could go to utils https://github.com/Exabyte-io/utils/blob/main/src/py/mat3ra/utils/array.py or somewhere there.

Reply:
In general, it seems that the logic you are using in this PR should be doable in a much more concise way - e.g. with a couple of numpy vector operations